### DISCUSSION TITLE
Extend GeneratorParams.SetSearchOption to allow setting strings

### DISCUSSION THREAD

**Discussion by koepalex** [microsoft/onnxruntime-genai#483] [New features / APIs]
First of all, awesome library, it is fun to use 👏 
It would be great extension to allow other types of search options for example string[].

Use case would be to pass `stop words` into the model (for example as defied by the [HuggingFace TextGenerationApi](https://huggingface.github.io/text-generation-inference/#/Text%20Generation%20Inference/compat_generate) parameter `stop`).

**Comment by natke** [comment]
Thank you for the suggestion @koepalex! I'm moving this into our new features area in Discussions
